### ISSUE TITLE
Suppress "explicit lifetime" and "descriptive name" warning

### ISSUE THREAD

**Issue by mverleg** [lalrpop/lalrpop#614]
I'm getting a lot of these warning:

```
warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
    --> .../out/grammar.rs:4106:1
     |
4106 | / fn __action56<
4107 | |     'input,
4108 | | >(
4109 | |     input: &'input str,
4110 | |     __0: (usize, alloc::vec::Vec<Stmt>, usize),
4111 | | ) -> Vec<Stmt>
     | |______________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
```

and 

```
warning: consider choosing a more descriptive name
    --> .../out/grammar.rs:3441:9
     |
3441 |     (_, __0, _): (usize, Vec<Stmt>, usize),
     |         ^^^
     |
     = note: `#[warn(clippy::just_underscores_and_digits)]` on by default
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#just_underscores_and_digits
```

**Comment by mgree** [comment]
Same! I tried putting `#![allow(clippy::...)]` in a few spots (in the `.lalrpop` file; before `lalrpop_mod!`), but nothing convinced it to be quiet.

**Comment by mgree** [comment]
Same! I tried putting `#![allow(clippy::...)]` in a few spots (in the `.lalrpop` file; before `lalrpop_mod!`), but nothing convinced it to be quiet. A full list of the warnings I get:

- clippy::wrong_self_convention
- clippy::too_many_arguments
- clippy::clone_on_copy 
- clippy::needless_lifetimes 
- clippy::type_complexity 
- clippy::just_underscores_and_digits

it would be convenient to turn these off just in the generated file, because I want them elsewhere!

**Comment by martin-t** [comment]
The `lalrpop_mod` macro accepts attributes so you can put them inside - e.g. `lalrpop_mod!(#[allow(clippy::all)] pub somename);`

The underlying issue is that #632 adds the `allow` into a submodule of the generated file, not the top - there is more code below this submodule which causes the warnings.

**Comment by mgree** [comment]
Ah, amazing... thank you!

**Comment by mverleg** [comment]
If I understand #632 correctly, this is fixed. Indeed I am not seeing the original warning locally.

**Comment by martin-t** [comment]
This doesn't seem to be fixed. I am on `0.19.8` (both `lalrpop` and `lalrpop-util`) and i am still getting these errors when running clippy.

Additionally, i don't see *how* it could be fixed, see my earlier comment - #632 puts `#![allow(clippy::all)]` inside the generated module but the `__actionXX` functions are generated outside the module so the attribute doesn't apply to them.

I don't know why you're no longer getting the warnings but I am still getting them in my project if i remove `#[allow(clippy::all)]` [here](https://github.com/martin-t/FML/blob/66991a428e26cdf7270f4c99bf35b3fbd200373e/src/main.rs#L10).
